app/api/api/views.py

```diff
@@ -515,6 +515,18 @@ def post(self, request, *args, **kwargs):
         content_type_str = body.pop("content_type", None)
         content_type = ContentType.objects.get(model=content_type_str)
         body["content_type"] = content_type.id
+        concept_id = body.get("concept", None)
```

Comment:
Could this section be moved down, to below the validation of whether a Concept exists? (Line `558`)

As you're then accessing the domain from the concept on line `519`, but it might not exist, so best to validate it.

This also removes the double lookup by to the Concepts table.